### ISSUE TITLE
[Blocks] Conditional Flow

### ISSUE THREAD

**Issue by aliabd** [gradio-app/gradio#738]
I would like to build a conditional flow based on output/selected input. 
 
Something like a radio input "pick one of these options" and based on choice show them one of two inputs (and run one of two functions). 

These leads to some cool demos:
* Multi step interfaces 
* Related interfaces that can handle different inputs (ie Text OR Image) https://github.com/gradio-app/gradio/issues/450

**Comment by FarukOzderim** [comment]
This feels like multiple tabs inside a tab, not sure it is suitable with our design. WDYT @pngwn and @aliabid94?

**Comment by abidlabs** [comment]
I agree that the use case @aliabd is describing can be accomplished through tabs. But there are similar use cases that can't easily be turned into a small number of tabs. For example, if you have a dropdown with 5-10 different options, and based on what is selected, you want to show them a particular interface. It would be too much to have 5-10 different tabs, but if you could _render_ a particular input depending on what has been selected, that would be pretty handy.

That being said, I can't think of a concrete use case right now, so perhaps we can deprioritize?

**Comment by abidlabs** [comment]
Basically the same as #784
